components/batch-system/src/batch.rs

```diff
@@ -471,6 +497,24 @@ impl<N: Fsm, C: Fsm, Handler: PollHandler<N, C>> Poller<N, C, Handler> {
                 batch.schedule(&self.router, r, false);
             }
         }
+        if let Some(fsm) = batch.control.take() {
+            self.router.control_box.release(fsm);
```

Comment:
Should be `self.router.control_scheduler.schedule(fsm)` as there may still have unhandled messages.  /cc @5kbpers